### ISSUE TITLE
GitHub Actions: Improvements to 'Automate In Progress Issue Updates: Add Update Label'

### ISSUE THREAD

**Issue by Aveline-art** [hackforla/website#2032]
### Dependency
- ~~#1977~~
- #2634 

### Overview
As a developer, we need to ensure that the GHA involving the 'To Update !' label creates clear comments and directions for the team to follow.

### Action Items

- [ ] Understand the code from the below directories.
- [ ] [JS Edits](https://github.com/hackforla/website/blob/gh-pages/github-actions/add-update-label-weekly/add-label.js)
  - [ ] ~~Add a new filter in the `isTimelineOutdated` function to see when an issue is assigned. If an issue is assigned more than 3 days ago, the issue is considered outdated, and the label is added.~~ ~~This may not be required once #2634 is merged.~~
  - [ ] ~~The timeline should indicate the timezone (which is set to PST). If possible, fix it in the `formatComment` function's `options` variable in the js file, rather than on the md file.~~
- [ ] [MD edits](https://github.com/hackforla/website/blob/gh-pages/github-actions/add-update-label-weekly/update-instructions-template.md)
  - [ ] In the md file, add instructions to put questions in the issue if help is needed with pictures if needed, and that these are examples of well-formed questions: https://github.com/hackforla/website/issues/1619#issuecomment-897315561 or https://github.com/hackforla/website/issues/1908#issuecomment-877908152
- [ ] Release dependency on #1976  

### Resources/Instructions
[add-update-label-weekly](https://github.com/hackforla/website/tree/gh-pages/github-actions/add-update-label-weekly)
[add-update-label-weekly.yml](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/add-update-label-weekly.yml)

#### Related Pages

#1956
#2006 

Never done GitHub actions? [Start here!](https://docs.github.com/en/actions)

Note that you might want to do something outside the scope of the above psudo-code. If so, be sure to leave comments in your PR or this issue that justifies your reasoning. If you feel you need guidance, be sure to reach out! We cannot foresee whether this issue is solvable, or what hard decisions have to be made, but we would love to hear and help you!

#### Additional resources:
[Events that trigger workflows](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)
[Workflow syntax for GitHub Actions](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions)
[GitHub GraphQL](https://docs.github.com/en/graphql)

**Comment by Aveline-art** [comment]
Bug:

#2040 is marked (today) because a timeline is outdated by default unless proven otherwise. Without a timeline to reference, we cannot check for outdatedness.

Possible solution:

Add a new filter to see when an issue is assigned.

**Comment by Aveline-art** [comment]
Bug:

The time does not indicate timezone.

**Comment by Aveline-art** [comment]
Note: the meeting times does not account for all teams. Change the text to something like, "ask for help at your next meeting".

**Comment by Aveline-art** [comment]
Note: set the previous comment to hide (perhaps by an invisible html comment marker to identify whether the comment is created by the GHA).

**Comment by Aveline-art** [comment]
Note: also explain that it helps to also put pictures and questions on your issue so that people can help you, otherwise, the team will be lost. Use this as a reference: https://github.com/hackforla/website/issues/1619#issuecomment-897315561 or https://github.com/hackforla/website/issues/1908#issuecomment-877908152

**Comment by Aveline-art** [comment]
Note: have instructions to swap the labels from 'To Update !' to "Status: updated'

**Comment by ExperimentsInHonesty** [comment]
@Aveline-art Option 1 please

> - [x] The GHA will add the 'To Update !' label and post a [request for update comment](https://github.com/hackforla/website/blob/gh-pages/github-actions/add-update-label-weekly/update-instructions-template.md).
>   - [ ] Pro: Will reduce the amount of scattered code for our developers to understand, and GHA logs for the tech lead to review.

**Comment by macho-catt** [comment]
Putting this as a dependency because we found an interesting issue for #1977 that may be related to this. Basically, if an issue was recently assigned but it falls under the timing window for asking to update an issue, it asks for an update even though it was recently picked up.

@SAUMILDHANKAR tagging you for your info

**Comment by ExperimentsInHonesty** [comment]
I  moved this out of the icebox and added the ready for dev lead labels because the dependencies are satisfied.

**Comment by ExperimentsInHonesty** [comment]
@t-will-gillis are you working on the issue or drafting it?  

- If you are drafting it, you can add a draft label and leave it in this column (new issue approval) while you complete the draft.  
- If you are working on it.  Then move it to in progress column

**Comment by angieyan** [comment]
Availability: Mon - Wed 9am - 11am
ETA: May 4, 23 EOD

**Comment by t-will-gillis** [comment]
Hi @angieyan - thanks for your interest in this issue. Please note that the **only** issues in the `Prioritized Backlog` column are eligible to be worked on. This issue was in the `New Issue Approval` column and is not available for you to work on. 

I will take care of us-assigning you from this issue and moving it back to the correct column. Please check out the [CONTRIBUTING.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md#23a-available-issues-for-new-members-front-endback-end) documents and available issues in the `Prioritized Backlog` column.

**Comment by t-will-gillis** [comment]
Note: This issue is being rewritten for these reasons:

- Since it is an older issue, some of the steps listed in the 'JS Edits' have already been addressed by other issues, e.g. #2634 and #3341 and the comments already display the time in PST. 
- The links to 'MD edits' and in 'Resources/ Instructions' are expired/ point to a 404 page.

Since deleting and/or striking the expired instructions will make this issue too muddled for clarity: The replacement issue is #4590

**Comment by angieyan** [comment]
@t-will-gillis at the end of the comments on #2652 issue you can see that Justin asked me if I could take #2032 issue, so I assigned this issue to me :)

**Comment by t-will-gillis** [comment]
@angieyan yes he did! I see that now. We have had others picking up issues from the wrong column before and I made a bad assumption. My apologies!!! 

I closed this issue and opened #4590 as a replacement- if you want to grab this one instead and make whatever changes as you see fit, I promise not to get in your way! Thank you for letting me know and apologies again...

**Comment by angieyan** [comment]
@t-will-gillis no worries :) Apologies accepted, but you need to help me understand how to test the action :)
